actors/builtin/paych/paych_test.go

```diff
@@ -23,158 +25,707 @@ func TestPaymentChannelActor_Constructor(t *testing.T) {
-			newPaychAddr,
+			paychAddr,
 			builtin.InitActorCodeID,
 			builtin.MultisigActorCodeID,
 			builtin.AccountActorCodeID,
 			exitcode.ErrIllegalArgument,
 		}, {"fails if sender (from) is not account actor",
-			newPaychAddr,
+			paychAddr,
 			builtin.InitActorCodeID,
 			builtin.MultisigActorCodeID,
 			builtin.AccountActorCodeID,
 			exitcode.ErrIllegalArgument,
 		}, {"fails if addr is not ID type",
-			tutil.NewActorAddr(t, "beach blanket babylon"),
+			tutil.NewSECP256K1Addr(t, "beach blanket babylon"),
 			builtin.InitActorCodeID,
-			builtin.MultisigActorCodeID,
+			builtin.AccountActorCodeID,
 			builtin.AccountActorCodeID,
 			exitcode.ErrIllegalArgument,
 		},
 	}
 	for _, tc := range testCases {
 		t.Run(tc.desc, func(t *testing.T) {
-			builder := mock.NewBuilder(ctx, newPaychAddr).
+			builder := mock.NewBuilder(ctx, paychAddr).
 				WithCaller(callerAddr, tc.callerCode).
-				WithActorType(newPaychAddr, tc.newActorCode).
+				WithActorType(tc.paymentChannelAddr, tc.newActorCode).
 				WithActorType(payerAddr, tc.payerCode)
 			rt := builder.Build(t)
 			rt.ExpectValidateCallerType(builtin.InitActorCodeID)
 			rt.ExpectAbort(tc.expExitCode, func() {
-				rt.Call(actor.Constructor, &ConstructorParams{To: newPaychAddr})
+				rt.Call(actor.Constructor, &ConstructorParams{To: tc.paymentChannelAddr})
 			})
 		})
 	}
+
+	t.Run("fails if actor does not exist with: no code for address", func(t *testing.T) {
+		builder := mock.NewBuilder(ctx, paychAddr).
+			WithCaller(callerAddr, builtin.InitActorCodeID).
+			WithActorType(payerAddr, builtin.AccountActorCodeID)
+		rt := builder.Build(t)
+		rt.ExpectValidateCallerType(builtin.InitActorCodeID)
+		rt.ExpectAbort(exitcode.ErrIllegalArgument, func() {
+			rt.Call(actor.Constructor, &ConstructorParams{To: paychAddr})
+		})
+	})
+}
+
+func TestPaymentChannelActor_CreateLaneSuccess(t *testing.T) {
+	ctx := context.Background()
+	rt, _, sv := requireCreateChannelWithLanes(t, ctx, 1)
+	var st State
+	rt.GetState(&st)
+	assert.Len(t, st.LaneStates, 1)
+	ls := st.LaneStates[0]
+	assert.Equal(t, sv.Amount, ls.Redeemed)
+	assert.Equal(t, sv.Nonce, ls.Nonce)
+	assert.Equal(t, sv.Lane, ls.ID)
 }
 
-func TestPaymentChannelActor_UpdateChannelState(t *testing.T) {
+func TestPaymentChannelActor_CreateLaneFailure(t *testing.T) {
 	ctx := context.Background()
 	actor := pcActorHarness{Actor{}, t}
 
-	newPaychAddr := tutil.NewIDAddr(t, 100)
-	callerAddr := tutil.NewIDAddr(t, 101)
+	initActorAddr := tutil.NewIDAddr(t, 100)
+	paychAddr := tutil.NewIDAddr(t, 101)
 	payerAddr := tutil.NewIDAddr(t, 102)
 
-	// set up trivial signature verifier and hasher
-	versig := func(sig crypto.Signature, signer addr.Address, plaintext []byte) bool { return true }
-	hasher := func(data []byte) [8]byte {
-		return [8]byte{}
+	sig := &crypto.Signature{Type: crypto.SigTypeBLS, Data: []byte("doesn't matter")}
+
+	testCases := []struct {
+		desc       string
+		targetCode cid.Cid
+
+		balance  int64
```

Comment:
`balance` looks unused.

Please factor out the `abi.NewTokenAmount(9)` below so it's encountered before the test cases (or use it here).